Adyen.EcommLibrary/Model/ApplicationInfo.cs

```diff
@@ -0,0 +1,187 @@
+        public CommonField AdyenPaymentSource { get; set; }
+
+        /// <summary>
+        /// Gets or Sets ExternalPlatform
+        /// </summary>
+        [DataMember(Name="externalPlatform", EmitDefaultValue=false)]
+        public ExternalPlatform ExternalPlatform { get; set; }
+
+        /// <summary>
+        /// Gets or Sets MerchantApplication
+        /// </summary>
+        [DataMember(Name="merchantApplication", EmitDefaultValue=false)]
+        public CommonField MerchantApplication { get; set; }
+
+        /// <summary>
+        /// Gets or Sets MerchantDevice
+        /// </summary>
+        [DataMember(Name="merchantDevice", EmitDefaultValue=false)]
+        public MerchantDevice MerchantDevice { get; set; }
+
+        /// <summary>
+        /// Gets or Sets ShopperInteractionDevice
+        /// </summary>
+        [DataMember(Name="shopperInteractionDevice", EmitDefaultValue=false)]
+        public ShopperInteractionDevice ShopperInteractionDevice { get; set; }
+
+        /// <summary>
+        /// Returns the string presentation of the object
+        /// </summary>
+        /// <returns>String presentation of the object</returns>
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            sb.Append("class ApplicationInfo {\n");
+            sb.Append("  AdyenLibrary: ").Append(AdyenLibrary).Append("\n");
+            sb.Append("  AdyenPaymentSource: ").Append(AdyenPaymentSource).Append("\n");
+            sb.Append("  ExternalPlatform: ").Append(ExternalPlatform).Append("\n");
+            sb.Append("  MerchantApplication: ").Append(MerchantApplication).Append("\n");
+            sb.Append("  MerchantDevice: ").Append(MerchantDevice).Append("\n");
+            sb.Append("  ShopperInteractionDevice: ").Append(ShopperInteractionDevice).Append("\n");
+            sb.Append("}\n");
+            return sb.ToString();
+        }
+  
+        /// <summary>
+        /// Returns the JSON string presentation of the object
+        /// </summary>
+        /// <returns>JSON string presentation of the object</returns>
+        public string ToJson()
+        {
+            return JsonConvert.SerializeObject(this, Formatting.Indented);
+        }
+
+        /// <summary>
+        /// Returns true if objects are equal
+        /// </summary>
+        /// <param name="input">Object to be compared</param>
+        /// <returns>Boolean</returns>
+        public override bool Equals(object input)
+        {
+            return this.Equals(input as ApplicationInfo);
+        }
+
+        /// <summary>
+        /// Returns true if ApplicationInfo instances are equal
+        /// </summary>
+        /// <param name="input">Instance of ApplicationInfo to be compared</param>
+        /// <returns>Boolean</returns>
+        public bool Equals(ApplicationInfo input)
+        {
+            if (input == null)
+                return false;
+
```

Comment:
Can we just add one more condition here to check if references are same before comparing fields?

Reply:
It is possible to remove these two methods.